src/nl/igorski/example/MWEngineActivity.java

```diff
@@ -326,7 +328,16 @@ public void handleNotification( int aNotificationId, int aNotificationValue )
             switch ( _notificationEnums[ aNotificationId ])
             {
                 case SEQUENCER_POSITION_UPDATED:
-                    Log.d( LOG_TAG, "sequencer position : " + aNotificationValue );
+
+                    // for this notification id, the notification value describes the current buffer position of the
+                    // engine, we can now calculate the current sequencer position (in steps) and the amount of
+                    // samples pending until the next step position is reached
+
+                    int sequencerPosition = ( int ) Math.floor( aNotificationValue / _sequencerController.getSamplesPerStep() );
+                    int nextPosition      = sequencerPosition + 1;
+                    int pendingSamples    = ( _sequencerController.getSamplesPerStep() * nextPosition ) - aNotificationValue;
+
+                    Log.d( LOG_TAG, "sequencer position:" + sequencerPosition + " samples elapsed:" + aNotificationValue + ", pending:" + pendingSamples );
                     break;
```

Comment:
Here we use the new methods from SequencerController to calculate the step position and pending samples (until the next position) from the current buffer position.